pandas/tests/indexing/test_at.py

```diff
@@ -47,6 +37,67 @@ def test_selection_methods_of_assigned_col():
     tm.assert_frame_equal(result, expected)
 
 
+@pytest.mark.parametrize(
+    "value_to_assign",
+    [
+        "a",
+        1,
+        1.0,
+        np.nan,
+        True,
+        ("a",),
+        ["a"],
+        datetime(2000, 1, 2, tzinfo=timezone.utc),
```

Comment:
As you can see the list of `value_to_assign` for a Series contains more items (in particular a dictionary, a numpy array and series). I tried to use them for a dataframe as well but was gettings errors. In particular:
- dictionary, series: `ValueError: Incompatible indexer with Series`
- numpy array: `IndexError:  only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) and integer or boolean arrays are valid indices`

I haven't looked more deeply in the code, but is this expected or should a bug be raised for it?

Reply:
it is likely we should not trying to treat iterables like this, so could be a bug. if you can open a separate issue for this (and PR would be great too!)

Reply:
What do you mean exactly by `not trying to treat iterables like this`? Do you mean that dictionaries, series, and numpy arrays shouldn't be assigned to dataframes / series?